### PR TITLE
Add Body Composition endpoint

### PR DESCRIPTION
**Pull request by jrbenito**
This returns a Json similar to:

```
{'totalAverage': {'bodyFatCount': 1, 'physiqueRating': None,
'muscleMass': 99999, 'weight': 99999.99908447266, 'boneMassCount': 1,
'bodyWaterCount': 1, 'visceralFat': None, 'bmiCount': 1, 'until':
1582070399999, 'bodyFat': 99.8700008392334, 'visceralFatCount': 0,
'bodyWater': 59.220001220703125, 'from': 1581984000000,
'muscleMassCount': 1, 'boneMass': 9999, 'metabolicAge': None,
'physiqueRatingCount': 0, 'weightCount': 1, 'bmi': 99.0,
'metabolicAgeCount': 0}, 'startDate': '2020-02-18', 'dateWeightList':
[{'weightDelta': -999.003662109375, 'physiqueRating': None,
'muscleMass': 99999, 'weight': 99999.99908447266, 'timestampGMT':
1582038680000, 'visceralFat': None, 'date': 1582009880000, 'bodyFat':
99.8700008392334, 'samplePk': 1582038680000, 'bodyWater':
99.220001220703125, 'sourceType': 'INDEX_SCALE', 'metabolicAge': None,
'boneMass': 9999, 'calendarDate': '2020-02-18', 'bmi': 99.0}],
'endDate': '2020-02-18'}
```

Signed-off-by: Josenivaldo Benito Jr <jrbenito@benito.qsl.br>

### CODE REVIEW

**Comment by cyberjunky** [comment]
Nice, but my stats are empty or zero, (weight: None for example) do you have any idea where this data is coming from?

**Comment by jrbenito** [comment]
Hi @cyberjunky ,

In my case this data come from a Garmin Index Scale (weight, muscle mass, bone mass, fat %, water %, bmi). Other stats like metabolic age, I have no idea (probably a device I do not own).

I believe that if you manual enter your weight in garmin connect app (weight in) or sync your weight through MyFitnessPal it shall work too.

**Comment by jrbenito** [comment]
For sake of documentation, there is another endpoint where you can get all your weight data:

https://connect.garmin.com/modern/proxy/userprofile-service/userprofile/personal-information/weightWithOutbound/ (all history data -> mine returns since 2014, when I used to weight in manually every day through the Garmin connect app)

https://connect.garmin.com/modern/proxy/userprofile-service/userprofile/personal-information/weightWithOutbound/filterByDay?from=1581797256000&until=1582038670000

in this later URL, filtered by from (epoch) until (epoch).

returns are like:

'''[
{
"samplePk": 1582038680000,
"date": "2020-02-18T07:11:20.000+0000",
"calendarDate": null,
"weight": 72960,
"bmi": 25,
"bodyFat": 18.8700008392334,
"bodyWater": 59.220001220703125,
"boneMass": 4590,
"muscleMass": 30520,
"physiqueRating": null,
"visceralFat": null,
"metabolicAge": null,
"caloricIntake": null,
"sourceType": null
},
{
"samplePk": 1581946776000,
"date": "2020-02-17T05:39:36.000+0000",
"calendarDate": null,
"weight": 73620,
"bmi": 25.200000762939453,
"bodyFat": 19.399999618530273,
"bodyWater": 58.83000183105469,
"boneMass": 4599,
"muscleMass": 30680,
"physiqueRating": null,
"visceralFat": null,
"metabolicAge": null,
"caloricIntake": null,
"sourceType": null
}
]'''

notice that source type in this endpoint comes as null even do I have garmin index scale and this data are actual valid data from myself.

**Comment by cyberjunky** [comment]
I have my weight manual in Garmin app, but the Body Composition output only seems to display the scale values. 
I was looking at buying one, but read some reviews about incorrect body mass, fat ratios, even between several readings in a row, are you experiencing this?
Will add your PR soon, it's a bit busy during weekdays.

**Comment by jrbenito** [comment]
The body mass is accurate for me, however the fat % swings a lot depending on what I ate last day and exercise. But, at least, it seems to swing for the right side. Like, if I don't exercise on weekend, drink a lot of beer and eat ton of barbecue it will swing up by 2% on monday and swing back 1% during the week if I keep doing well.
While I have no idea how accurate is its in that regarding, muscle mass don't swing like that (what I know should not) and weight is accurate enough

**Comment by Etragas** [comment]
My impression from what I've read is that the body fat % number is generally incorrect, but systematically so. I.e if you lose weight it should trend downwards. (I'm a recent garmin index purchaser and keen on the features in this PR!)

File: README.md [hunk 1]
```diff
@@ -27,31 +27,35 @@ from garminconnect import (
...
+    GarminConnectAuthenticationError,
+    GarminConnectTooManyRequestsError,
+) as err:
+    print("Error occured during Garmin Connect Client setup: %s", err)
+    return
+except Exception:  # pylint: disable=broad-except
+    print("Unknown error occured during Garmin Connect Client setup")
+    return
+
+#"""Get Full name"""
```
Comment (Etragas) [inline README.md]: I think switching from switching from """ to a comment makes sense, but then shouldn't we drop the """?

Comment (jrbenito) [reply README.md]: > My impression from what I've read is that the body fat % number is generally incorrect, but systematically so. I.e if you lose weight it should trend downwards. (I'm a recent garmin index purchaser and keen on the features in this PR!)

That is my impression too. I don't have a easy way to measure body fat% accuracy (probably go to a physicist and take a test, weight in same day before eat or drink and compare?). But to be honest, I don't care if the thing is very precise in body fat or muscle readings, but do care about weight. If other numbers give me a ballpark to work with, that's enough to my needs. Just need to know if I am going in the right direction ;)

Comment (jrbenito) [reply README.md]: > I think switching from switching from """ to a comment makes sense, but then shouldn't we drop the """?

That sounds good, but let @cyberjunky tell us what he wants to do.

**Pull request merged**